Sources/ResourceProvider/Operators/Concurrent.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  Concurrent.swift
+//  swift-resource-provider
+//
+//  Created by Óscar Morales Vivó on 10/19/24.
+//
+
+private struct ConcurrentSyncProvider<P: SyncProvider>: AsyncProvider where P: Sendable {
```

Comment:
nothing in this type looks concurrent? I'm a bit confused.

Reply:
It is concurrent in that it can be called concurrently, nothing more than that.

This is as opposed to the serialized version which is an `actor`.

Noting that all these types exist mostly for debuggability. Stack traces were indecipherable otherwise.